datahub-graphql-api/src/main/java/com/linkedin/datahub/graphql/resolvers/query/DatasetResolver.java

```diff
@@ -0,0 +1,31 @@
+package com.linkedin.datahub.graphql.resolvers.query;
+
+import com.linkedin.datahub.graphql.generated.Dataset;
+import com.linkedin.datahub.graphql.loaders.DatasetLoader;
+import com.linkedin.datahub.graphql.mappers.DatasetMapper;
+import com.linkedin.datahub.graphql.resolvers.AuthorizedResolver;
+
+import graphql.schema.DataFetchingEnvironment;
+import org.dataloader.DataLoader;
+import java.util.concurrent.CompletableFuture;
+
+import static com.linkedin.datahub.graphql.Constants.*;
+
+
+/**
+ * Resolver responsible for resolving the 'dataset' field of the Query type
+ */
+public class DatasetResolver extends AuthorizedResolver<CompletableFuture<Dataset>> {
+    @Override
+    protected boolean isAuthorized(DataFetchingEnvironment environment) {
+        // Always authorize reads, for now. Override this resolver to customize authorization.
```

Comment:
Please file an issue so we don't forget

Reply:
This was actually meant to be a simple example of AuthorizedResolver in the absence of any mutations. I don't think we need to authorize the action of reading a particular dataset, authentication should suffice. 

Maybe I'll just make this an AuthenticatedResolver and leave AuthorizedResolver unused for now.